policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/stream/SerDesConfig.java

```diff
@@ -0,0 +1,146 @@
+ * In general, the keys are not used so the choice of serialiser is not important
+ */
+public final class SerDesConfig {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private static final String SERIALIZATION_FAILED_MESSAGE = "Failed to serialize ";
+    private static final String DESERIALIZATION_FAILED_MESSAGE = "Failed to deserialize ";
+
+    private SerDesConfig() {
+        // Static collection of objects, class should never be instantiated
+    }
+
+    /**
+     * Kafka key serialiser for upstream messages coming in as input
+     * Used by the Rest Controller to insert requests onto the topic
+     *
+     * @return an appropriate key serialiser for the topic's message content
+     */
+    public static Serializer<String> resourceKeySerializer() {
+        return new StringSerializer();
+    }
+
+    /**
+     * Kafka value serialiser for upstream messages coming in as input
+     * Used by the Rest Controller to insert requests onto the topic
+     *
+     * @return an appropriate value serialiser for the topic's message content (AttributeMaskingRequest)
+     */
+    public static Serializer<PolicyRequest> resourceValueSerializer() {
+        return (String ignored, PolicyRequest policyRequest) -> {
+            try {
+                return MAPPER.writeValueAsBytes(policyRequest);
+            } catch (IOException e) {
+                throw new SerializationFailedException(SERIALIZATION_FAILED_MESSAGE + policyRequest.toString(), e);
+            }
+        };
+    }
+
+    /**
+     * Kafka key deserialiser for upstream messages coming in as input
+     *
+     * @return an appropriate key deserialiser for the topic's message content
+     */
+    public static Deserializer<String> resourceKeyDeserializer() {
+        return new StringDeserializer();
+    }
+
+    /**
+     * Kafka value deserialiser for upstream messages coming in as input
+     *
+     * @return an appropriate value deserialiser for the topic's message content (AttributeMaskingRequest)
+     */
+    public static Deserializer<PolicyRequest> resourceValueDeserializer() {
+        return (String ignored, byte[] policyRequest) -> {
+            try {
+                return MAPPER.readValue(policyRequest, PolicyRequest.class);
+            } catch (IOException e) {
+                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + new String(policyRequest, Charset.defaultCharset()), e);
+            }
+        };
+    }
+
+    /**
+     * Kafka key serialiser for downstream messages going out as output
+     *
+     * @return an appropriate key serialiser for the topic's message content
+     */
+    public static Serializer<String> ruleKeySerializer() {
+        return new StringSerializer();
+    }
+
+    /**
+     * Kafka value serialiser for downstream messages going out as output
+     *
+     * @return an appropriate value serialiser for the topic's message content (AttributeMaskingResponse)
```

Comment:
AttributeMaskingRequest in JavaDoc should be PolicyRequest

Reply:
removed